src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -62,7 +64,13 @@ public class SnowflakeSinkTask extends SinkTask {
   // snowflake
   // account and execute queries
   private SnowflakeConnectionService conn = null;
-  private String id = "-1";
+
+  // tracks number of tasks the config wants to create
+  private String taskConfigId = "-1";
+  // tracks total number of tasks created in this kc instance
+  private static int taskCreationCount = 0;
+  // tracks number of times this task instance has been started
+  private int taskInstanceStartCount;
```

Comment:
I am a bit confused on the usage of these two. 
Since `taskCreationCount` is static variable, I am assuming it will keep on incrementing until JVM dies. When will this happen? 

taskInstanceStartCount - seems will be incremented every time after rebalance?

Reply:
I would recommend hitting few Connect REST APIs to verify correctness. 
There is a restart task API https://docs.confluent.io/platform/current/connect/references/restapi.html#tasks

And there are two more connector APIs: pause, stop

Reply:
taskCreationCount (now totalTaskCreationCount) is to show how many tasks have been created in the KC instance. I originally assumed that it would be cleared when KC is killed, but that's a bad assumption to make. So i've added extra calls in KC start/stop to reset it to 0.

taskInstanceStartCount (now taskOpenCount) is meant to count the number of times the task has been opened, to hopefully show the number of times this task has been rebalanced. I didn't realize that start is only called once, whereas open is called during rebalance (i believe?)